block.py

```diff
@@ -59,11 +59,14 @@ def setup(self):
         self.add_blockdev_cmd(cmds, blacklist='ram.*')
 
         lsblk = self.collect_cmd_output("lsblk -f -a -l")
-        # for LUKS devices, collect cryptsetup luksDump
+        # for LUKS devices, collect cryptsetup, luksDump and clevis
+        # if a LUKS device exists, ensure /etc/crypttab is collected
         if lsblk['status'] == 0:
             for line in lsblk['output'].splitlines():
                 if 'crypto_LUKS' in line:
                     dev = line.split()[0]
                     self.add_cmd_output('cryptsetup luksDump /dev/%s' % dev)
+                    self.add_cmd_output('clevis luks list -d /dev/%s' % dev)
+                    self.add_copy_spec('/etc/crypttab')
```

Comment:
I'm not following why this needs to be in the for loop?

If there are no LUKS devices, there should be either not be an `/etc/crypttab` or it should be empty, right? `add_copy_spec()` gracefully handles when a requested file doesn't exist, and copying an empty/template file doesn't seem all that bad to me.

Is there some condition causing us to _not_ want to collect this file if it exists, yet no LUKS devices are present?

Reply:
Yes, you are right - there won't be an /etc/crypttab if no LUKS devices are present. I figured inserting it into the loop would be a clean way to keep the block module compact and the LUKS data capture contained in a single loop. We could of course move this outside of the LUKS loop and just grab it anyway - just wanted to minimize the chances of pulling any irrelevant data.  I can see why it's redundant as it would pull the file for every LUKS device found; but as you say it does handle situations like that gracefully - so I would defer to your direction here :)